CMakeLists.txt

```diff
@@ -13,6 +13,10 @@ if(DEFINED ENV{VCPKG_ROOT} AND NOT DEFINED CMAKE_TOOLCHAIN_FILE)
   set(CMAKE_TOOLCHAIN_FILE
       "$ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
       CACHE STRING "")
+else()
+  # set default toolchain to vcpkg submodule
+  set(CMAKE_TOOLCHAIN_FILE ${CMAKE_CURRENT_SOURCE_DIR}/vcpkg/scripts/buildsystems/vcpkg.cmake
+    CACHE STRING "Vcpkg toolchain file")
```

Comment:
I understand the desire to do this. I would make it easier for new devs, but it is essentially locking everyone into using vcpkg.
Things like `CMAKE_TOOLCHAIN_FILE` are and `CMAKE_CONFIGURATION_TYPES` should not be forced.
Isn't checking for vcpkg in the environment variables good enough?